extension/src/tdigest.rs

```diff
@@ -245,6 +247,89 @@ CREATE AGGREGATE timescale_analytics_experimental.tdigest(size int, value DOUBLE
+pub fn tdigest_compound_trans(
+    state: Option<Internal<InternalTDigest>>,
+    value: Option<timescale_analytics_experimental::TDigest<'static>>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal<InternalTDigest>> {
+    unsafe {
+        in_aggregate_context(fcinfo, || {
+            match (state, value) {
+                (a, None) => a,
+                (None, Some(a)) => Some(a.to_internal_tdigest().into()),
+                (Some(a), Some(b)) => {
+                    Some(InternalTDigest::merge_digests(
+                            vec![a.deref().clone(), b.to_internal_tdigest()]  // TODO: TDigest merge with self
+                        ).into())
+                }
+            }
+        })
+    }
+}
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+pub fn tdigest_compound_combine(
+    state1: Option<Internal<InternalTDigest>>,
+    state2: Option<Internal<InternalTDigest>>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal<InternalTDigest>> {
+    unsafe {
+        in_aggregate_context(fcinfo, || {
+            match (state1, state2) {
+                (None, None) => None,
+                (None, Some(state2)) => Some(state2.clone().into()),
+                (Some(state1), None) => Some(state1.clone().into()),
+                (Some(state1), Some(state2)) => {
+                    Some(InternalTDigest::merge_digests(
+                            vec![state1.deref().clone(), state2.deref().clone()]  // TODO: TDigest merge with self
+                        ).into())
+                }
+            }
+        })
+    }
+}
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+fn tdigest_compound_final(
+    state: Option<Internal<InternalTDigest>>,
+    _fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<timescale_analytics_experimental::TDigest<'static>> {
+    match state {
+        None => None,
+        Some(state) => Some(TDigest::from_internal_tdigest(&state.deref())),
+    }
+}
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+fn tdigest_compound_serialize(
+    state: Internal<InternalTDigest>,
+    _fcinfo: pg_sys::FunctionCallInfo,
+) -> bytea {
+    crate::do_serialize!(state)
+}
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+pub fn tdigest_compound_deserialize(
+    bytes: bytea,
+    _internal: Option<Internal<()>>,
+) -> Internal<InternalTDigest> {
+    crate::do_deserialize!(bytes, InternalTDigest)
+}
+
+extension_sql!(r#"
+CREATE AGGREGATE timescale_analytics_experimental.tdigest(
+    timescale_analytics_experimental.tdigest
+) (
+    sfunc = timescale_analytics_experimental.tdigest_compound_trans,
+    stype = internal,
+    finalfunc = timescale_analytics_experimental.tdigest_compound_final,
+    combinefunc = timescale_analytics_experimental.tdigest_compound_combine,
+    serialfunc = timescale_analytics_experimental.tdigest_compound_serialize,
+    deserialfunc = timescale_analytics_experimental.tdigest_compound_deserialize
```

Comment:
Need to add parallel = safe?

Reply:
Ah, thanks for the reminder.  Added to both aggregates here and also in uddsketch.